folly/detail/Clock.cpp

```diff
@@ -48,7 +48,125 @@ int clock_getres(clockid_t clk_id, struct timespec* ts) {
 
   return 0;
 }
-#elif _MSC_VER
+#elif defined(_MSC_VER)
+// The MSVC version has been extracted from the pthreads implemenation here:
+// https://github.com/songdongsheng/libpthread
+// Copyright(c) 2011, Dongsheng Song <songdongsheng@live.cn>
+//
+// It is under the Apache License Version 2.0, just as the rest of the file is.
```

Comment:
This is the most permissive license of a proper implementation of this that I could find. It should be the same as the rest of the file, unless I missed something? Re-implementing it is possible, as trying to work around not having it will be more of a pain, as it's used in a few spots in HHVM.